src/panel.tsx

```diff
@@ -48,12 +68,15 @@ export function Panel() {
   return React.useMemo(() => {
     if (!shareDescriptor) return null;
 
-    const embedUrl = `${ABSTRACT_APP_URL}/embed/${inferShareId(
-      shareDescriptor
-    )}?find={id}`;
+    let url = new URL(`/embed/${inferShareId(shareDescriptor)}`, ABSTRACT_APP_URL);
```

Comment:
```suggestion
    const url = new URL(shareDescriptor.url ||  `/embed/${inferShareId(shareDescriptor)}`, ABSTRACT_APP_URL);
```

It’s either going to have url or id, but not both. Also means we don’t need to do any sort of merging 😊